conu/backend/docker/image.py

```diff
@@ -221,6 +221,51 @@ def push(self, repository=None, tag=None):
                                         self.name, error)
         return image
 
+    def copy(self, repository, tag="latest", source_transport_index=2, target_transport_index=2):
+        """
+        WIP
+        Copy image to repository:tag
+        :param repository to be copied to
+        :param tag
+        :param source_transport_index in transport list WIP. for testing
+                can I find this out tranport from DockerImage instance? maybe useless param
+        :param target_transport_index in tranposrts list WIP, for testing
+        :return: the new DockerImage
+        """
+        transports = ["containers-storage:",
+                      "dir:",
+                      "docker://",
+                      "docker-archive",
+                      "docker-daemon:",
+                      "oci:",
+                      "ostree:"]
+
+        # doing things in local space, maid be deleted later
+        def transport_param(index, repo, tagg):
+            """
+            Parses info into skopeo parameter
+            :param index: index in transports list, kinda temporary
+            :param repo:
+            :param tagg:
+            :return: string. skopeo parameter specifying image
+            """
+            if not 0 <= index < len(transports):
+                raise ValueError("Invalid source transport index: " + str(index))
+            # command=transports[index]
+            if index in [0, 2, 4, 5]:
+                return transports[index] + repo + ":" + tagg
+            raise NotImplementedError(transports[index] + "transport is not implemented yet")
+
+        run_cmd(["skopeo", "copy",
+                 transport_param(source_transport_index, self.name, self.tag),
+                 transport_param(target_transport_index, repository, tag)])
+
+        # questionable.. repo is for example palusus/alpine..
+        # tried to pull this to local "docker-daemon"..
+        # it got pulled to docker.io/palusus/alpine instead
```

Comment:
this something skopeo (and the underlaying go libraries) do on their own: they prefix the image name with `<registry>/<namespace>/...` if it's not set originally; just treat as a feature of skopeo, not a bug